src/librustdoc/html/static/js/search.js

```diff
@@ -1382,16 +1411,330 @@ function initSearch(rawSearchIndex) {
+                const hofInputs = fnType.generics;
+                pushText(fnType, result);
+                pushText({name: " (", highlighted: false}, result);
+                let needsComma = false;
+                for (const fnType of hofInputs) {
+                    if (needsComma) {
+                        pushText({ name: ", ", highlighted: false }, result);
+                    }
+                    needsComma = true;
+                    writeFn(fnType, result);
+                }
+                pushText({
+                    name: hofOutput.length === 0 ? ")" : ") -> ",
+                    highlighted: false,
+                }, result);
+                if (hofOutput.length > 1) {
+                    pushText({name: "(", highlighted: false}, result);
+                }
+                needsComma = false;
+                for (const fnType of hofOutput) {
+                    if (needsComma) {
+                        pushText({ name: ", ", highlighted: false }, result);
+                    }
+                    needsComma = true;
+                    writeFn(fnType, result);
+                }
+                if (hofOutput.length > 1) {
+                    pushText({name: ")", highlighted: false}, result);
+                }
+            }
+            /**
+             * Write a type. This function checks for special types,
+             * like slices, with their own formatting. It also handles
+             * updating the where clause and generic type param map.
+             *
+             * @param {FunctionType} fnType
+             * @param {[string]} result
+             */
+            function writeFn(fnType, result) {
+                if (fnType.id < 0) {
+                    const queryId =  mgens && mgens.has(fnType.id) ? mgens.get(fnType.id) : null;
+                    if (fnParamNames[-1 - fnType.id] === "") {
+                        for (const nested of fnType.generics) {
+                            writeFn(nested, result);
+                        }
+                        return;
+                    } else if (queryId < 0) {
+                        mappedNames.set(
+                            fnParamNames[-1 - fnType.id],
+                            queryParamNames[-1 - queryId],
+                        );
+                    }
+                    pushText({
+                        name: fnParamNames[-1 - fnType.id],
+                        highlighted: !!fnType.highlighted,
+                    }, result);
+                    const where = [];
+                    let first = true;
+                    for (const nested of fnType.generics) {
+                        if (first) {
+                            first = false;
+                        } else {
+                            pushText({ name: " + ", highlighted: false }, where);
+                        }
+                        writeFn(nested, where);
+                    }
```

Comment:
It drives me nuts that we can't skip the first n elements in a `for` loop in JS... Maybe the loop should be rewritten to handle this case (code is shorter too :smiley:):

```suggestion
                    for (let i = 1; len = fnType.generics.length; i < len; ++i) {
                        pushText({ name: " + ", highlighted: false }, where);
                        writeFn(fnType.generics[i], where);
                    }   
```

Reply:
I just realized that this case was present in a few places. So might be worth writing a function:

```js
function pickAName(array, notFirstElementCall, everyElementsCall) {
    if (array.length !== 0) {
        everyElementsCall(array[0]);
        for (let i = 0, len = array.length; i < len; ++i) {
            const elem = array[i];
            notFirstElementCall(elem);
            everyElementsCall(elem);
        }
    }
}
```

Then we could replace the loop with:

```js
pickAName(fnType.generics, elem => writeFn(elem, where), elem => pushText({ name: " + ", highlighted: false }, where));
```

Well, tell me what you think about it.

Reply:
The `bindings` map?